File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/service/DeleteCashService.java
package com.mondiamedia.ahmedbadr.githubreopos.service;


import androidx.annotation.NonNull;

import com.firebase.jobdispatcher.JobParameters;
import com.firebase.jobdispatcher.JobService;

import io.realm.Realm;

public class DeleteCashService extends JobService {

    private Realm mRealm;

    @Override
    public void onCreate() {
        super.onCreate();
        mRealm = Realm.getDefaultInstance();
    }

    @Override
    public boolean onStartJob(@NonNull JobParameters job) {
        mRealm.executeTransaction(Realm::deleteAll);
        return false;
    }

    @Override
    public boolean onStopJob(@NonNull JobParameters job) {
        return false;
    }
}

File: app/src/main/java/com/mondiamedia/ahmedbadr/githubreopos/service/DeleteCashService.kt
package com.mondiamedia.ahmedbadr.githubreopos.service


import com.firebase.jobdispatcher.JobParameters
import com.firebase.jobdispatcher.JobService

import io.realm.Realm

class DeleteCashService : JobService() {

    private lateinit var realm: Realm

    override fun onCreate() {
        super.onCreate()
        realm = Realm.getDefaultInstance()
    }

    override fun onStartJob(job: JobParameters): Boolean {
        realm.executeTransaction { it.deleteAll() }
        return false
    }

    override fun onStopJob(job: JobParameters): Boolean {
        return false
    }
}
